open-sphere-base/core/src/main/java/io/opensphere/core/pipeline/processor/PointSpriteProcessor.java

```diff
@@ -346,15 +346,17 @@ private void resetProjectionSensitiveGeometries(Viewer view, ViewChangeSupport.V
             {
                 myLastHandledHeading = heading;
 
-                List<PointSpriteGeometry> projectionSensitiveGeoms = getGeometries().stream()
-                        .filter(g -> g.isProjectionSensitive()).collect(Collectors.toList());
-                if (CollectionUtilities.hasContent(projectionSensitiveGeoms))
+                // When re-processing rotated images, we want to ensure they
+                // aren't otherwise modified.
+                synchronized (getGeometrySet())
                 {
-                    for (PointSpriteGeometry geom : projectionSensitiveGeoms)
-                    {
-                        getImageManagersToGeoms().remove(geom.getImageManager());
-                    }
-                    resetDueToImageUpdate(New.linkedList(projectionSensitiveGeoms), State.UNPROCESSED);
+                    List<PointSpriteGeometry> projectionSensitiveGeoms = getGeometries().parallelStream()
+                            .filter(g -> g.isProjectionSensitive()).collect(Collectors.toCollection(New::list));
+
+                    projectionSensitiveGeoms.parallelStream()

```

Comment:
I dont think this needs to be a parrall stream becuase removing the image manger is not a costly task to do and if our projectionSensitiveGeom is small, seems like the overhead to create a parallelStream() is overkill